Confusion between `CURL*` and `CURLM*`

**What component of `google-cloud-cpp` is this related to?**
`google/cloud/storage`

**Describe the bug**
I think there is a confusion between `CURL*` and `CURLM*` here
https://github.com/googleapis/google-cloud-cpp/blob/f4ea7beb523742c649977908bf83f50d12fae4a0/google/cloud/storage/internal/curl_handle_factory.cc#L127

**To Reproduce**
Compiling `google/cloud/storage` with `--copt=-msse4.2 --copt=-mavx --compilation_mode=opt`

I get this error even after fixing our code:

```
ERROR: /usr/local/google/home/coryan/.cache/bazel/_bazel_coryan/f1b649d8eddab70da217bfe95503f270/external/upb/BUILD:68:11: C++ compilation of rule '@upb//:upb' failed (Exit 1): gcc failed: error executing command /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffunction-sections ... (remaining 26 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox gcc failed: error executing command /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffunction-sections ... (remaining 26 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
In file included from /usr/include/string.h:495,
                 from external/upb/upb/upb.h:16,
                 from external/upb/upb/upb.c:2:
In function 'strncpy',
    inlined from 'upb_status_seterrmsg' at external/upb/upb/upb.c:40:3:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:106:10: error: '__builtin_strncpy' specified bound 127 equals destination size [-Werror=stringop-truncation]
  106 |   return __builtin___strncpy_chk (__dest, __src, __len, __bos (__dest));
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
INFO: Elapsed time: 3.950s, Critical Path: 3.03s
```

That is coming from the upb library: https://github.com/protocolbuffers/upb

--- a/google/cloud/storage/internal/curl_handle_factory.cc
+++ b/google/cloud/storage/internal/curl_handle_factory.cc
@@ -91,7 +91,7 @@ PooledCurlHandleFactory::~PooledCurlHandleFactory() {
 CurlPtr PooledCurlHandleFactory::CreateHandle() {
   std::unique_lock<std::mutex> lk(mu_);
   if (!handles_.empty()) {
-    CURL* handle = handles_.back();
+    auto* handle = handles_.back();
     // Clear all the options in the handle so we do not leak its previous state.
     (void)curl_easy_reset(handle);
     handles_.pop_back();
@@ -112,7 +112,7 @@ void PooledCurlHandleFactory::CleanupHandle(CurlHandle&& h) {
     last_client_ip_address_ = ip;
   }
   if (handles_.size() >= maximum_size_) {
-    CURL* tmp = handles_.front();
+    auto* tmp = handles_.front();
     handles_.erase(handles_.begin());
     curl_easy_cleanup(tmp);
   }
@@ -124,7 +124,7 @@ void PooledCurlHandleFactory::CleanupHandle(CurlHandle&& h) {
 CurlMulti PooledCurlHandleFactory::CreateMultiHandle() {
   std::unique_lock<std::mutex> lk(mu_);
   if (!multi_handles_.empty()) {
-    CURL* m = multi_handles_.back();
+    auto* m = multi_handles_.back();
     multi_handles_.pop_back();
     return CurlMulti(m, &curl_multi_cleanup);
   }
@@ -134,7 +134,7 @@ CurlMulti PooledCurlHandleFactory::CreateMultiHandle() {
 void PooledCurlHandleFactory::CleanupMultiHandle(CurlMulti&& m) {
   std::unique_lock<std::mutex> lk(mu_);
   if (multi_handles_.size() >= maximum_size_) {
-    CURLM* tmp = multi_handles_.front();
+    auto* tmp = multi_handles_.front();
     multi_handles_.erase(multi_handles_.begin());
     curl_multi_cleanup(tmp);
   }

